src/utilities/layer-from-json.ts

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable prettier/prettier */
+import { layerTypes } from '../layer';
+import { ILayer, ILayerJSON } from '../layer/base-layer';
+
+export default function layerFromJSON(
+  jsonLayer: ILayerJSON,
+  ...args: unknown[]
+): ILayer {
+  const Layer = layerTypes[jsonLayer.type];
+
+  const realLayer = new Layer(...args);
```

Comment:
Again, `Layer` has `any` type :( We might run into issue if layer type is `Internal` (which is static)

Reply:
Yeah, I knew that would come up :). Sorry, but I'm a little confused about how I'm supposed to handle it. I know that at its core, the layer type can be an `ILayer` or `InternalModel`. Some types have constructors, others don't. So how do I construct layers that don't have constructors?

Reply:
I think we might need to have an `enum` collection for all layer types. (already available in `layer.type` property). Then we can call appropriate constructor based on the comparison.

Reply:
Sorry I'm not very familiar with this project, but would be the appropriate constructor?

Reply:
>Some types have constructors, others don't. 

That shouldn't be the case.  Even if a layer is compound (ie, consisting of several internal layers) they are all translated to several layers.  An example is from: https://github.com/BrainJS/brain.js/blob/f572a3d2489950cf530d7f7cee7721167744d351/src/layer/feed-forward.ts#L9

feed-forward layer is translated to:
* Random
* Random
* Multiply
* Add
* Sigmoid

Reply:
Oh, I was looking at layers such as `Internal` and `InternalModel`.